pom.xml

```diff
@@ -41,34 +42,44 @@
 
     <build>
         <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-dependency-plugin</artifactId>
-                <executions>
-                <execution>
-                    <id>unpack-dependencies</id>
-                    <phase>prepare-package</phase>
-                    <goals>
-                        <goal>unpack-dependencies</goal>
-                    </goals>
-                    <configuration>
-                        <excludes>META-INF/**</excludes>
-                        <outputDirectory>${project.build.directory}/classes</outputDirectory>
-                        <overWriteReleases>false</overWriteReleases>
-                        <overWriteSnapshots>true</overWriteSnapshots>
-                        <includeArtifactIds>osgi.core,commons-text,commons-lang3,org.apache.sling.feature,org.osgi.util.function,org.apache.felix.cm.json,org.apache.sling.commons.johnzon,org.apache.felix.converter,commons-cli,slf4j-api,slf4j-simple</includeArtifactIds>
+                <plugin>
+                    <groupId>biz.aQute.bnd</groupId>
+                    <artifactId>bnd-maven-plugin</artifactId>
+                        <configuration>
+                        <bnd><![CDATA[
+                 Main-Class: org.apache.sling.feature.launcher.impl.Main
+                -exportcontents:\
+                 org.apache.sling.feature.launcher.spi.*
+                -conditionalpackage:\
+                 javax.json*,\
+                 org.apache.commons.cli.*,\
+                 org.apache.commons.lang3.*,\
+                 org.apache.commons.text.*,\
+                 org.apache.felix.cm.json.*,\
+                 org.apache.felix.converter.*,\
+                 org.apache.johnzon.core.*,\
+                 org.apache.johnzon.core.util.*,\
+                 org.apache.sling.feature.*,\
+                 org.osgi.*,\
```

Comment:
I don't think we should package org.osgi classes - nor should we import them. The launcher shouldn't have dependencies on org.osgi.

Reply:
I like to have the bundle-meta-data in a manifest. Together with the smaller build that is the only reason.

I just packaged everything that was in before. Just because of -conditionalpackage the bundle is 400kb smaller than before!

We are using [BundleStartLevel](https://github.com/apache/sling-org-apache-sling-feature-launcher/blob/14fd3f7476214baf2ee114e71d50f6518fd75fd8/src/main/java/org/apache/sling/feature/launcher/impl/launchers/AbstractRunner.java#L323) and the static field `FrameworkEvent.STOPPED_SYSTEM_REFRESHED` in bootstrap.

not sure how to remove the `BundleStartLevel`

Reply:
The two classes in the launcher package that use osgi core api are loaded via reflection when the real class path is assembled - hence, they should get their packages from the framework. The static field should be inlined, no? If not, we could consider using the value directly. 

I think we need the  org.osgi.util.function for the  org.osgi.util.converter - but we hopefully don't need the rest.

Reply:
I guess you would now need to tell bnd to not import org.osgi packages as well.

Reply:
would me nice to (later) get rid of some of the commons dependencies. Just used for `StringSubstitutor`

Reply:
Right, for now, I think this looks good but please remove the imports for org.osgi from the headers.

Reply:
is removed org.osgi from imports

Reply:
I think you only removed it from the -conditionalpackage. The manifest still looks like this for me: 
```
Import-Package: java.io,java.lang,java.lang.annotation,java.lang.invoke,
 java.lang.reflect,java.math,java.net,java.nio,java.nio.charset,java.nio
 .file,java.nio.file.attribute,java.security,java.text,java.util,java.ut
 il.concurrent,java.util.concurrent.atomic,java.util.function,java.util.
 jar,java.util.logging,java.util.regex,java.util.stream,java.util.zip,ja
 vax.script,javax.xml.xpath,org.osgi.framework;version="[1.9,2)",org.osg
 i.framework.launch;version="[1.2,2)",org.osgi.framework.startlevel;vers
 ion="[1.0,2)",org.osgi.resource;version="[1.0,2)",org.osgi.util.tracker
 ;version="[1.5,2)",org.xml.sax
```

I don't think we need the import for org.osgi.framework,org.osgi.framework.launch,org.osgi.framework.startlevel,,org.osgi.resource,org.osgi.util.tracker 

Additionally, I wonder if we need the import for javax.script - that seems odd. Not sure about javax.xml.xpath and org.xml.sax. That might not be needed either....

Reply:
I will have a look why this imports are there..
but for example `org.osgi.framework.launch` is used in AbstractRunner and FrameworkRunner. So why not to import?

Reply:
As I said:

> The two classes in the launcher package that use osgi core api are loaded via reflection when the real class path is assembled - hence, they should get their packages from the framework. The static field should be inlined, no? If not, we could consider using the value directly.
> 
> I think we need the org.osgi.util.function for the org.osgi.util.converter - but we hopefully don't need the rest.

So, for the org.osgi packages, that is something that bnd detects but is not needed (which it can't know). Otherwise, it would be a mistake to not conditional package them :-).

For the other imports, I'm not sure they are needed - feels like they are not but I'm not sure....

Reply:
I removed the osgi imports .
`Import-Package: javax.script,javax.xml.xpath,org.xml.sax`